test_close_session_signal intermittently failing

I'm seeing intermittent test failures with 0.8.0 ([example full log](https://buildd.debian.org/status/fetch.php?pkg=libportal&arch=ppc64el&ver=0.8.0-1&stamp=1725366680&raw=0)) which still seem to be present in 0.8.1:

```
=================================== FAILURES ===================================
_________________ TestRemoteDesktop.test_close_session_signal __________________

self = <pyportaltest.test_remotedesktop.TestRemoteDesktop testMethod=test_close_session_signal>

    def test_close_session_signal(self):
        """
        Ensure that we get the GObject signal when our session is closed
        externally.
        """
        params = {"close-after-start": 500}
        setup = self.create_session(params=params)
        session = setup.session
    
        session_closed_signal_received = False
    
        def session_closed(session):
            nonlocal session_closed_signal_received
            session_closed_signal_received = True
            self.mainloop.quit()
    
        session.connect("closed", session_closed)
    
        self.mainloop.run()
    
>       assert session_closed_signal_received is True
E       assert False is True

pyportaltest/test_remotedesktop.py:599: AssertionError
```

I'm not sure I understand how this can happen: if `session_closed` didn't run, then how can we leave `self.mainloop.run` without having set `session_closed_signal_received`?

Is it possible that `self.mainloop` might sometimes be terminated by a leftover signal handler from a previous test, or something like that?

--- a/tests/pyportaltest/test_remotedesktop.py
+++ b/tests/pyportaltest/test_remotedesktop.py
@@ -56,6 +56,7 @@ class TestRemoteDesktop(PortalTest):
         start_session=True,
         persist_mode=None,
         restore_token=None,
+        cb_closed_signal=None,
     ) -> SessionSetup:
         params = params or {}
         # To make the tests easier, load ScreenCast automatically if we have
@@ -119,6 +120,9 @@ class TestRemoteDesktop(PortalTest):
         session_handle = options["session_handle_token"]
 
         assert session.get_session_type() == Xdp.SessionType.REMOTE_DESKTOP
+
+        if cb_closed_signal:
+            session.connect("closed", cb_closed_signal)
 
         if outputs:
             # open the PW remote by default since we need it anyway for Start()
@@ -581,9 +585,6 @@ class TestRemoteDesktop(PortalTest):
         Ensure that we get the GObject signal when our session is closed
         externally.
         """
-        params = {"close-after-start": 500}
-        setup = self.create_session(params=params)
-        session = setup.session
 
         session_closed_signal_received = False
 
@@ -592,8 +593,10 @@ class TestRemoteDesktop(PortalTest):
             session_closed_signal_received = True
             self.mainloop.quit()
 
-        session.connect("closed", session_closed)
-
+        params = {"close-after-start": 500}
+        _ = self.create_session(
+            params=params, cb_closed_signal=session_closed
+        )  # keep SessionResult alive
         self.mainloop.run()
 
         assert session_closed_signal_received is True

